app/lib/track/src/track.cpp

```diff
@@ -269,3 +272,58 @@ Track::Track(const char *f1, const char *f2, AudioInputI2S *s)
       busRightToRecording(busRight, 0, recording, 1), readFileName(f1),
       writeFileName(f2){};
 #endif
+
+// changes read file from library selection
+void Track::setWriteFileName(String name){
+  int Length = name.length() + 1;
+
+  Serial.print(Length);
+  char newReadFile[Length];
+  
+  name.toCharArray(newReadFile, Length);
+  Serial.printf("Read file name was: %s\n", readFileName);
+  this->readFileName = newReadFile;
+  Serial.printf("Read file name now: %s\n", readFileName);
+  Serial.printf("Write file name: %s\n", writeFileName);
+ 
+}
```

Comment:
The `readFileName` and `writeFileName` values should stay constant, and when a loop is loaded, it should be copied over to the file labelled by `readFileName`, not `writeFileName`. Currently, the program would crash if the same loop was loaded into two separate tracks. Also, copying the file allows the original to remain as a backup.